langwatch_nlp/langwatch_nlp/studio/execute/execute_flow.py

```diff
@@ -43,30 +43,42 @@ async def execute_flow(
         True if event.manual_execution_mode is None else event.manual_execution_mode
     )
 
+    do_not_trace = event.do_not_trace
+
     disable_dsp_caching()
 
     # TODO: handle workflow errors here throwing an special event showing the error was during the execution of the workflow?
     yield start_workflow_event(workflow, trace_id)
 
     try:
-        with langwatch.trace(
-            trace_id=event.trace_id,
-            api_key=event.workflow.api_key,
-            skip_root_span=True,
-            metadata={
-                "platform": "optimization_studio",
-                "environment": "development" if manual_execution_mode else "production",
-            },
-        ) as trace:
-            trace.autotrack_dspy()
-
+        if do_not_trace:
             module = WorkflowModule(
                 workflow,
                 manual_execution_mode=manual_execution_mode,
                 until_node_id=until_node_id,
                 inputs=inputs[0] if inputs else None,
             )
             module.set_reporting(queue=queue, trace_id=trace_id, workflow=workflow)
+        else:
+            with langwatch.trace(
+                trace_id=event.trace_id,
+                api_key=event.workflow.api_key,
+                skip_root_span=True,
+                metadata={
+                    "platform": "optimization_studio",
+                    "environment": (
+                        "development" if manual_execution_mode else "production"
+                    ),
+                },
+            ) as trace:
+                trace.autotrack_dspy()
+                module = WorkflowModule(
+                    workflow,
+                    manual_execution_mode=manual_execution_mode,
+                    until_node_id=until_node_id,
+                    inputs=inputs[0] if inputs else None,
+                )
+                module.set_reporting(queue=queue, trace_id=trace_id, workflow=workflow)
```

Comment:
no no, this is wrong, you are not tracking the whole execution inside the trace anymore, look at the previous identeation

if you want to do this, you need to extract everything that was inside the `as trace:` to a separate function, or (maybe easier), you could have a `optional_langwatch_trace` context wrapper function, something like that:

```python
@contextmanager
def optional_langwatch_trace(do_not_trace=False, trace_id, api_key, skip_root_span, metadata):
  if do_not_trace:
    yield None
  else:
    with langwatch.trace(blablalba) as trace:
      yield trace
```